packages/dynamoose/lib/Model/index.ts

```diff
@@ -104,7 +105,7 @@ export interface ModelTableOptions extends TableOptionsOptional {
 
 interface ModelInternalProperties {
 	name: string;
-	options: TableOptionsOptional;
+	options: ModelTableOptions;
```

Comment:
What is the purpose of this change?

Reply:
interface ModelTableOptions extends TableOptionsOptional and is [a type of Model.contructor thrird argument "options"](https://github.com/dynamoose/dynamoose/pull/1594/files/88309e897166d0df225b4f4e9ef21bbc7e164f41#diff-1d6224e89f76a90eae897c943da376eda750143ddf3e19912f4beaba46768893R194)

Since ModelInternalProperties.options used to store this "options" value, it will be correct if it has the same type.

Reply:
@famence But why is it necessary? Why does `TableOptionsOptional` not work in this case?